internal/controller/conditionsutil/conditions_util_test.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2024 the Pinniped contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package conditionsutil
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"go.pinniped.dev/internal/plog"
+)
+
+func TestMergeIDPConditions(t *testing.T) {
+	twoHoursAgo := metav1.Time{Time: time.Now().Add(-2 * time.Hour)}
+	oneHourAgo := metav1.Time{Time: time.Now().Add(-1 * time.Hour)}
+	testTime := metav1.Now()
+
+	tests := []struct {
+		name               string
+		newConditions      []*metav1.Condition
+		conditionsToUpdate *[]metav1.Condition
+		observedGeneration int64
+		wantResult         bool
+		wantLogSnippets    []string
+		wantConditions     []metav1.Condition
+	}{
+		{
+			name: "True -> False returns true",
```

Comment:
So this means:

"An existing condition that was true but became false will return true"?

Reply:
This is both testing MergeIDPConditions as well as HasErrorConditions.

So naming of the tests should be clear.

Reply:
Let's move this over to https://github.com/vmware-tanzu/pinniped/issues/1915